src/transformers/utils/bitsandbytes.py

```diff
@@ -164,17 +166,18 @@ def replace_with_bnb_linear(model, modules_to_not_convert=None, current_key_name
                                 compress_statistics=quantization_config.bnb_4bit_use_double_quant,
                                 quant_type=quantization_config.bnb_4bit_quant_type,
                             )
+                            has_been_replaced = True
                     # Force requires grad to False to avoid unexpected errors
                     model._modules[name].requires_grad_(False)
         # Remove the last key for recursion
         if len(list(module.children())) > 0:
-            replace_with_bnb_linear(
+            _, has_been_replaced = replace_with_bnb_linear(
                 module,
                 modules_to_not_convert,
                 current_key_name,
                 quantization_config,
             )
-    return model
+    return model, has_been_replaced
```

Comment:
I think we should make the recursive `replace_with_bnb_linear` private and wrap it in a new `replace_with_bnb_linear` that would raise the warning. How does that sound?

Reply:
Sounds great!